node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -185,18 +185,25 @@ impl OrderingProvider {
 	) -> Result<()> {
 		if let Some(activated) = update.activated.as_ref() {
 			// Fetch ancestors of the activated leaf.
-			let ancestors = self
-				.get_block_ancestors(sender, activated.hash, activated.number)
-				.await
-				.unwrap_or_else(|err| {
-					tracing::debug!(
-						target: LOG_TARGET,
-						activated_leaf = ?activated,
-						"Skipping leaf ancestors due to an error: {}",
-						err
-					);
-					Vec::new()
-				});
+			let finalized_block_number = get_finalized_block_number(sender).await?;
+
+			let ancestors = Self::get_block_ancestors(
+				sender,
+				activated.hash,
+				activated.number,
+				finalized_block_number,
+				&mut self.last_observed_blocks,
+			)
```

Comment:
We should limit the number of blocks we query at once, if we have a stalling finally, we don't want to run into OOM executions or exorbitant block times because all wasm instances are consumed

Reply:
We are not using the runtime to get this information. IIUC, get_block_ancestors() this makes use of the substrate client to do this which should rely on the local db.

Reply:
Hmm, good point!